optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -15,180 +15,204 @@
-
-    mixed_qkv = self.Wqkv(hidden_states)
-    if self.clip_qkv:
-        mixed_qkv = mixed_qkv.clamp(min=-self.clip_qkv, max=self.clip_qkv)
-
-    bs, seq_len, three_times_hidden_size = mixed_qkv.shape
-    mixed_qkv = mixed_qkv.view(bs, seq_len, self.n_heads * 3, self.head_dim)
-    mixed_qkv = mixed_qkv.transpose(1, 2)
-    query_states, key_states, value_states = (
-        mixed_qkv[:, : self.n_heads, ...],
-        mixed_qkv[:, self.n_heads : 2 * self.n_heads, ...],
-        mixed_qkv[:, 2 * self.n_heads :, ...],
-    )
-
-    if past_key_value is not None:
-        if len(past_key_value) != 0:
-            if token_idx is not None:
-                past_key_value[0].index_copy_(2, token_idx - 1, key_states)
-                past_key_value[1].index_copy_(2, token_idx - 1, value_states)
-                key_states = past_key_value[0]
-                value_states = past_key_value[1]
-            else:
-                key_states = torch.cat([past_key_value[0], key_states], dim=2)
-                value_states = torch.cat([past_key_value[1], value_states], dim=2)
-                past_key_value = [key_states, value_states]
-    else:
-        past_key_value = [
-            torch.empty(key_states.shape, dtype=key_states.dtype, device=key_states.device),
-            torch.empty(key_states.shape, dtype=key_states.dtype, device=key_states.device),
-        ]
-        past_key_value[0][:] = key_states[:]
-        past_key_value[1][:] = value_states[:]
-
-    query_length = seq_length if past_key_value is None else seq_length + past_key_value[0].shape[2]
-
-    if position_bias is not None:
-        if len(position_bias.shape) != 3:
-            raise ValueError(f"Expecting position_bias shape to be 3 dimensions, got {len(position_bias.shape)}")
-        key_length = key_states.shape[-2]
-
-        position_bias_query_index = max(0, position_bias.size(1) - query_length)
-        position_bias_key_index = max(0, position_bias.size(2) - key_length)
-
-        position_bias = position_bias[:, position_bias_query_index:, position_bias_key_index:]
-
-    if use_flash_attention and FusedSDPA:
-        import habana_frameworks.torch.hpu as ht
-
-        with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
-            attn_output = FusedSDPA.apply(
-                query_states,
-                key_states,
-                value_states,
-                attention_mask * torch.finfo(query_states.dtype).min + position_bias.to(query_states.dtype),
-                0.0,
-                False,
-                None,
-            )
-        attn_weights = None
-    else:
-        attention_scores = torch.matmul(query_states, key_states.transpose(-1, -2)) * self.softmax_scale
+logger = logging.get_logger(__name__)
+
+
+class Softmax(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x, dim=None, invAttnHead=None):
+        return torch.ops.hpu.softmax_fp8(x, dim, None, None, invAttnHead)
```

Comment:
Since INC is enabled, please use `torch.nn.functional.softmax` as its supported module by INC for quantization.
https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-functions
```suggestion
        return torch.nn.functional.softmax(x, dim)
```

Reply:
updated the code with your suggestion.